### PR TITLE
fix(prefer-wait-for): only report when imported from TL

### LINKED ISSUE #151: prefer-wait-for rule autofix is modifying Cypress wait() functions
Hello,

I have tried to add the rule 'prefer-wait-for' to my code but an unexpected interaction is provoked with my Cypress files. The autofix from the rule is also changing the calls to cy.Wait() for cy.WaitFor() in my Cypress test files

Screenshot 2020-06-09 at 14 04 13

### PR DESCRIPTION
**Pull request by timdeschryver**
Closes #151

### CODE REVIEW

File: lib/rules/prefer-wait-for.ts [hunk 1]
```diff
@@ -119,36 +115,44 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
...
 
-        importNodes.forEach(importNode => {
-          reportImport(importNode);
+          context
+            .getDeclaredVariables(importSpecifier)
+            .forEach(variable =>
+              variable.references.forEach(reference =>
+                reportWait(reference.identifier)
+              )
+            );
```
Comment (timdeschryver) [inline lib/rules/prefer-wait-for.ts:136]: Is there a way we could make this cleaner? 👀

File: tests/lib/rules/prefer-wait-for.test.ts [hunk 2]
```diff
@@ -41,6 +41,25 @@ ruleTester.run(RULE_NAME, rule, {
...
+
+      async () => {
+        await foo.wait();
+      }`,
+    },
+    {
+      code: `
+      cy.wait();
+      `,
+    },
```
Comment (gndelia) [inline tests/lib/rules/prefer-wait-for.test.ts:62]: I think we should consider testing `require` scenarios as well

File: lib/rules/prefer-wait-for.ts (hunk 1 above)
Comment (gndelia) [reply lib/rules/prefer-wait-for.ts:136]: perhaps

```ts
context
  .getDeclaredVariables(importSpecifier)
  .flatMap(variable => variable.references)
  .forEach(({ identifier })=> reportWait(identifier))
```

? But IMO it's not too much of a difference - I think this is fine

And I think flatMap does not work in node <= 10

Comment (Belco90) [reply lib/rules/prefer-wait-for.ts:136]: Indeed, we can't use [flatMap in node 10](https://node.green/#ES2019-features-Array-prototype--flat--flatMap-)

Comment (Belco90) [reply lib/rules/prefer-wait-for.ts:136]: Maybe we could make the cod cleaner by extracting the first forEach callback to its own function? It's the only thing I can think of.

File: tests/lib/rules/prefer-wait-for.test.ts (hunk 2 above)
Comment (Belco90) [reply tests/lib/rules/prefer-wait-for.test.ts:62]: It would be nice if we can have an additional test for it, indeed.

**Comment by timdeschryver** [comment]
This won't be just additional tests, it would also mean that we have to modify the rule.
e.g. the following is currently valid (but invalid with `import`):

```ts
const { wait } = require("@testing-library/foo");

async () => {
  await wait();
}
```

**Comment by Belco90** [comment]
Ah of course. Fair enough, I'll approve it then.

**Comment by Belco90** [comment]
@all-contributors please add @miguelerja for bug

**Pull request merged**